### DISCUSSION TITLE
Any tool suggestion for git add -i

### DISCUSSION THREAD

**Discussion by ccoVeille** [MitMaro/git-interactive-rebase-tool#905] [Q&A] [answered]
Hi @MitMaro 

I love your tool and I'm using it for more than a year I think.

You saved me hours of works. I love the fact you added support for automatic update-refs.

I'm curious, your tool provided an amazing solution for rebase interactive, do you have plan to work on git-interactive-add-tool ? Or equivalent.

I would like to know if there is something as good as your tool that could be used for adding files, or partial like the way `git rebase -i` does but as convenient as your tool.

I already know, love and use forgit (forgit::add) https://github.com/wfxr/forgit maintained by @wfxr but I'm curious if you know something else or plan to work on it.

**Comment by MitMaro** [comment, marked as answer]
> I love your tool and I'm using it for more than a year I think.
> 
> You saved me hours of works. I love the fact you added support for automatic update-refs.

Thank you for those kind comments! I'm happy that you found the tool useful and that it is saving your hours of work!

> I'm curious, your tool provided an amazing solution for rebase interactive, do you have plan to work on git-interactive-add-tool ? Or equivalent.

As much as I think this would be a great little tool and project, I don't think I have the time to take on another project.

It could be something that is integrated into this project that could take advantage of a lot of the code contained that has already been written, but at that point it's starting to be more like [Magit](https://github.com/magit/magit/blob/main/README.md). It would also require a significant rewrite of some of the internals.

A related idea I had toyed with in the past was to allow a set of diff hunks to be "yanked" from the current commit, and a new commit to be added with those hunks. This would essentially allow for splitting a commit during rebase. But this isn't really what you are looking. 😅

**Reply by ccoVeille** [reply]
Thanks for your reply.

I understand that. Make it wider would bring more issues, I'm able to use [forgit](https://github.com/wfxr/forgit) to do what I said, so it's ok.

I was curious and you replied my question.

**Comment by ccoVeille** [comment]
I found https://github.com/andrewshadura/git-crecord 

I'll have to test it and check but @andrewshadura might be what I was looking for.

**Comment by ccoVeille** [comment]
I also found that tig supports such a mode

https://www.reddit.com/r/git/comments/1gqfacq/graphical_git_add_p/

**Comment by ccoVeille** [comment]
Found this today, very promising

https://github.com/cwarden/git-add--interactive

cc @cwarden (thanks BTW)
